include/OpenColorIO/OpenColorIO.h

```diff
@@ -3080,7 +3088,11 @@ class OCIOEXPORT GpuShaderDesc : public GpuShaderCreator
 {
 public:
 
-    /// Create the legacy shader description.
+    /**
+     * Create the legacy GPU shader description. This is now deprecated.
+     * Do not use it because that's broken. So, this method has been replaced by the pair 
+     * Processor::getOptimizedLegacyGPUProcessor() and GpuShaderDesc::CreateShaderDesc().
+     */
     static GpuShaderDescRcPtr CreateLegacyShaderDesc(unsigned edgelen);
```

Comment:
Could be a good place to use [[deprecated]] on C++ 14 compiler and beyond.

Reply:
Something I will investigate. I also found a good example in the `Imath` project.
```
//
// Simple way to mark things as deprecated.
// When we are sure that C++14 is our true minimum, then we can just
// directly use [[deprecated(msg)]].
//
#if defined(_MSC_VER)
# define IMATH_DEPRECATED(msg) __declspec(deprecated(msg))
#elif __cplusplus >= 201402L
# define IMATH_DEPRECATED(msg) [[deprecated(msg)]]
#elif defined(__GNUC__) || defined(__clang__)
# define IMATH_DEPRECATED(msg) __attribute__((deprecated(msg)))
#else
# define IMATH_DEPRECATED(msg) /* unsupported on this platform */
#endif
```

Reply:
After few investigations, I propose to do the change in a dedicated pull request for `master` branch only. 

This pull request is planned to be integrated in the `RB-2.0` branch upon TSC approval. But adding the macro triggers lot of warnings in the Python bindings which should not be changed in the 2.0 release. It can only be changed in the `master` because that's the location for the coming 2.1 release.

Reply:
Worth noting here that we intend to remove this function in v2.1.0?

Reply:
Done